tracer/src/Datadog.Trace/Debugger/Conditions/ProbeCondition.cs

```diff
@@ -0,0 +1,143 @@
+// <copyright file="ProbeCondition.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq.Expressions;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Debugger.Conditions
+{
+    /// <summary>
+    /// Method phase
+    /// </summary>
+    public enum MethodPhase
+    {
+        /// <summary>
+        /// Entry of the method
+        /// </summary>
+        Entry,
+
+        /// <summary>
+        /// End of the method
+        /// </summary>
+        End
+    }
+
+    internal class ProbeCondition
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(ProbeCondition));
+        private readonly List<ScopeMember> _scopeMembers;
+
+        public ProbeCondition(string conditionId, MethodPhase evaluateAt, string conditionJson)
+        {
+            Id = conditionId;
+            EvaluateAt = evaluateAt;
+            ConditionJson = conditionJson;
+            _scopeMembers = new List<ScopeMember>();
+            Condition = new Lazy<Condition>(() => ProbeConditionExpressionParser.ToCondition(ConditionJson, InvocationTarget, ScopeMembers));
+        }
+
+        public Lazy<Condition> Condition { get; set; }
+
+        public bool ShouldEvaluate { get; set; }
+
+        public string Id { get; }
+
+        // we can save this as T but it will require native changes to return MethodDebuggerState<T> instead MethodDebuggerState
+        public ScopeMember InvocationTarget { get; private set; }
+
+        public MethodPhase EvaluateAt { get; }
+
+        public string ConditionJson { get; }
```

Comment:
This seems like an unused property? Do we really need to keep the `ConditionJson` around after we've parsed it?

Reply:
Only for error logs in case we failed to run the predicate. do you think we should get rid of it?